frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -421,6 +440,7 @@ export const insightLogic = kea<insightLogicType>([
         ],
     })),
     reducers(({ props }) => ({
+        query: [getDefaultQuery() as Node, { setQuery: (_, { query }) => query }],
```

Comment:
Good luck! Making everything rely ono this feels like it'll bring a lot of pain 😅.